sunpy/physics/differential_rotation.py

```diff
@@ -87,317 +78,73 @@ def diff_rot(duration, latitude, rot_type='howard', frame_time='sidereal'):
+    ( 0.,  7.24822784,  1.00695436)>): (Tx, Ty, distance) in (arcsec, arcsec, km)
+    (-562.37689548,  119.26840368,   1.50083152e+08)>
     """
 
-    # must have pairs of co-ordinates
-    if np.array(x).shape != np.array(y).shape:
-        raise ValueError('Input co-ordinates must have the same shape.')
-
     # Make sure we have enough time information to perform a solar differential
-    # rotation
-    # Start time
-    dstart = parse_time(tstart)
-    dend = parse_time(tend)
-    interval = (dend - dstart).total_seconds() * u.s
-
-    # Get the Sun's position from the vantage point at the start time
-    vstart = kwargs.get("vstart", _calc_P_B0_SD(dstart))
-    # Compute heliographic co-ordinates - returns (longitude, latitude). Points
-    # off the limb are returned as nan
-    longitude, latitude = convert_hpc_hg(x.to(u.arcsec).value,
-                                         y.to(u.arcsec).value,
-                                         b0_deg=vstart["b0"].to(u.deg).value,
-                                         l0_deg=vstart["l0"].to(u.deg).value,
-                                         dsun_meters=(constants.au * sun.sunearth_distance(t=dstart)).value,
-                                         angle_units='arcsec')
-    longitude = Longitude(longitude, u.deg)
-    latitude = Angle(latitude, u.deg)
-    # Compute the differential rotation
-    drot = diff_rot(interval, latitude, frame_time=frame_time,
-                    rot_type=rot_type)
-
-    # Convert back to heliocentric cartesian in units of arcseconds
-    vend = kwargs.get("vend", _calc_P_B0_SD(dend))
-
-    # It appears that there is a difference in how the SSWIDL function
-    # hel2arcmin and the sunpy function below performs this co-ordinate
-    # transform.
-    newx, newy = convert_hg_hpc(longitude.to(u.deg).value + drot.to(u.deg).value,
-                                latitude.to(u.deg).value,
-                                b0_deg=vend["b0"].to(u.deg).value,
-                                l0_deg=vend["l0"].to(u.deg).value,
-                                dsun_meters=(constants.au * sun.sunearth_distance(t=dend)).value,
-                                occultation=False)
-    newx = Angle(newx, u.arcsec)
-    newy = Angle(newy, u.arcsec)
-    return newx.to(u.arcsec), newy.to(u.arcsec)
-
-
-def _calc_P_B0_SD(date):
-    """
-    To calculate the solar P, B0 angles and the semi-diameter as seen from
-    Earth.  This function is assigned as being internal as these quantities
-    should be calculated in a part of SunPy that can calculate these quantities
-    accurately.
-
-    Parameters
-    -----------
-    date : `sunpy.time.time`
-        the time at which to calculate the solar P, B0 angles and the
-        semi-diameter.
-
-    Returns
-    -------
-    A dictionary with the following keys with the following meanings:
+    # rotation.
+    if coordinate.obstime is None:
```

Comment:
I wonder if you actually need to catch this here, the transformation will error when you try it?